sageopt/tests/test_coniclifts/test_affine_operators.py

```diff
@@ -106,3 +106,129 @@ def test_trace_and_diag(self):
         expr0 = aff.trace(temp)
         expr1 = aff.sum(x)
         assert Expression.are_equivalent(expr0, expr1)
+
+    def test_block(self):
+        A = np.eye(2) * 2
+        B = np.eye(3) * 3
+        expected = np.block([
+            [A, np.zeros((2, 3))],
+            [np.ones((3, 2)), B]
+        ])
+        actual = aff.block([
+            [A, np.zeros((2, 3))],
+            [np.ones((3, 2)), B]
+        ])
+        assert np.allclose(expected, actual)
```

Comment:
For these various stacking/concatenation/splitting tests, could you include some coniclifts Expressions in the tests? 

I'm thinking that a quick fix would be to pick one of the numpy arrays (here, maybe ``A``) and set a coniclifts Variable to have that value. A modified version of this test could be
```
A = np.eye(2) * 2
A_cl = Variable(shape=A.shape)
A_cl.value = A
B = np.eye(3) * 3
expected = np.block([
    [A, np.zeros((2, 3))],
    [np.ones((3, 2)), B]
])
actual = aff.block([
    [A_cl, np.zeros((2, 3))],
    [np.ones((3, 2)), B]
])
assert np.allclose(expected, actual.value)
A_cl.value = 1 + A  # or 2*A, or really anything different from A itself.
assert not np.allclose(expected, actual.value)
```
The two differences are that ``actual`` is now a coniclifts Expression depending on the coniclifts Variable ``A_cl``. The first assert statement now check that the numeric value of the Expression ``actual`` is what we expect, and the second assert statement makes sure that updates to ``A_cl`` propogate to the Expression called ``actual``.

Reply:
Sounds good, I'll model the remainder of the affine.py tests based on your code example above.

Reply:
Thanks!